lib/services/profiles-api-service.js

```diff
@@ -75,32 +75,29 @@ function profileApiServiceFactory(
         return _.flattenDeep([macs]);
     };
 
-    ProfileApiService.prototype.setLookup = function(req, res) {
-        var query = req.swagger ? req.swagger.query : req.query;
+    ProfileApiService.prototype.getRequestMacAddress = function(query, ipAddress) {
+        if (query.macs && query.ips) {
+            var macAddresses = _.flattenDeep([query.macs]);
+            var ipAddresses = _.flattenDeep([query.ips]);
 
-        if (!query.macs || !query.ips)
-            return Promise.resolve();
-
-        /* Select request's mac and ip, then set them*/
-        var macAddresses = _.flattenDeep([query.macs]);
-        var ipAddresses = _.flattenDeep([query.ips]);
-
-        var index = _.findIndex(ipAddresses, function(ip) {
-            return (ip && (ip === res.locals.ipAddress));
-        });
+            var index = _.findIndex(ipAddresses, function(ip) {
+                return (ip && (ip === ipAddress));
+            });
 
-        if (index < 0 || !macAddresses[index]) {
-            return Promise.resolve();
+            if(index >= 0) {
```

Comment:
missing !macAddresses[index] check

Reply:
The function is designed to return `undefined/null` when we can't get macAddress.